abstract_specification.md

```diff
@@ -233,68 +298,298 @@ Any given storage "engine" upon which SOMA is implemented may have additional fe
-| get(string key)                  | Get the object associated with the key                                                   |
-| has(string key)                  | Test for the existence of key in collection.                                             |
-| set(string key, ValueType value) | Set the key/value in the collection.                                                     |
-| del(string key)                  | Remove the key/value from the collection. Does not delete the underlying object (value). |
-| iterator                         | Iterate over the collection.                                                             |
-| get length                       | Get the number of elements in the collection.                                            |
+| Operation                                           | Description                                                                              |
+| --------------------------------------------------- | ---------------------------------------------------------------------------------------- |
+| get(string key)                                     | Get the object associated with the key.                                                  |
+| has(string key)                                     | Test for the existence of key in collection.                                             |
+| iterator                                            | Iterate over the collection.                                                             |
+| get length                                          | Get the number of elements in the collection.                                            |
+| set(string key, SOMAObject value, use_relative_uri) | Set the key/value in the collection.                                                     |
+| del(string key)                                     | Remove the key/value from the collection. Does not delete the underlying object (value). |
+| add_new_collection(string key, ...)                 | Creates a new sub-Collection and adds it to this Collection.                             |
+| add_new_dataframe(string key, ...)                  | Creates a new DataFrame and adds it to this Collection.                                  |
+| add_new_dense_ndarray(string key, ...)              | Creates a new DenseNDArray and adds it to this Collection.                               |
+| add_new_sparse_ndarray(string key, ...)             | Creates a new SparseNDArray and adds it to this Collection.                              |
+
+A SOMA collection also manages the lifecycle of objects directly instantiated by it.
+Objects accessed via getting a Collection element, or objects created with one of the <code>add_new\_<var>object_type</var></code> methods are considered "owned" by the collection.
+All such objects will be automatically closed together by [the collection's close operation](#operation-close-collection-types).
```

Comment:
Also worth mentioning they inherit the parent's context? Or do we consider that implementation-specific?